File: api/app/Models/Trip.php
<?php

namespace App\Models;

use App\Casts\PgArray;
use Database\Factories\TripFactory;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

/**
 * Le trajet RÉCURRENT du conducteur.
 *
 * departure_time reste une string "HH:MM:SS" — jamais de cast datetime :
 * un trajet récurrent n'a pas de date, un Carbon complet ici est un bug.
 * Les places libres se calculent par date via bookings, jamais en
 * décrémentant seats_total.
 */
class Trip extends Model
{
    /** @use HasFactory<TripFactory> */
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var list<string>
     */
    protected $fillable = [
        'driver_id',
        'origin_label',
        'origin_point',
        'dest_label',
        'dest_point',
        'route',
        'departure_time',
        'duration_minutes',
        'days_of_week',
        'seats_total',
        'price_per_seat',
        'active',
    ];

    /**
     * Get the attributes that should be cast.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'days_of_week' => PgArray::class,
            'active' => 'boolean',
        ];
    }

    public function driver(): BelongsTo
    {
        return $this->belongsTo(User::class, 'driver_id');
    }

    public function bookings(): HasMany
    {
        return $this->hasMany(Booking::class);
    }
}

File: app/lib/features/trip/domain/trip.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'trip.freezed.dart';
part 'trip.g.dart';

/// Le trajet récurrent tel que l'API le renvoie. C'EST le modèle.
@freezed
abstract class Trip with _$Trip {
  const factory Trip({
    required int id,
    required String originLabel,
    required String destLabel,
    /// « 06:45 » — un trajet récurrent n'a pas de date.
    required String departureTime,
    required int durationMinutes,
    required List<int> daysOfWeek,
    required int seatsTotal,
    required int pricePerSeat,
    required bool active,
  }) = _Trip;

  factory Trip.fromJson(Map<String, dynamic> json) => _$TripFromJson(json);
}
